### PR TITLE
Solution page

### LINKED ISSUE #63: challenge - display solution code

### PR DESCRIPTION
**Pull request by hgranthorner**
closes #63

### CODE REVIEW

File: client/components/Solution.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      <h2>Our solution</h2>
+      <div className="d-flex justify-content-around row">
+        <div className="col">
+          <CodeMirror defaultValue={individualChallenge.html} options={options} />
+        </div>
+        <div className="col">
+          <CodeMirror defaultValue={individualChallenge.css} options={options} />
+        </div>
+        <div className="col">
+          <CodeMirror defaultValue={individualChallenge.js} options={options} />
```
Comment (theo333) [inline client/components/Solution.js]: I thought we were going to remove the JS editor?
```suggestion
```

File: client/components/Solution.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      <h2>Your solution</h2>
+      <div className="d-flex justify-content-around row">
+        <div className="col">
+          <CodeMirror defaultValue={userchallenge.html} options={options} />
+        </div>
+        <div className="col">
+          <CodeMirror defaultValue={userchallenge.css} options={options} />
+        </div>
+        <div className="col">
+          <CodeMirror defaultValue={userchallenge.js} options={options} />
```
Comment (theo333) [inline client/components/Solution.js]: I thought we were going to remove the JS editor?
```suggestion
```

File: client/store/userchallenge.js [hunk 3]
```diff
@@ -20,6 +20,13 @@ export const updateUserchallenge = (userAnswer, userchallengeId, isSubmit) => di
...
 };
 
+export const fetchUserchallengeById = id => dispatch => {
+  return axios
+    .get(`/api/userchallenges/${id}`)
+    .then(res => res.data)
+    .then(userchallenge => dispatch(setUserchallenge(userchallenge)));
+};
+
 export const fetchUserchallenge = (userId, challengeId) => dispatch => {
```
Comment (theo333) [inline client/store/userchallenge.js]: Is your `fetchUserchallengeById` replacing `fetchUserchallenge`?  Seems like is doing the same thing, except your route does not include `userId`.  Your route is cleaner.

Comment (hgranthorner) [reply client/store/userchallenge.js]: Fetching user challenge takes a user id and a challenge id and fetches the active user challenge for that challenge. Mine is just grabbing a specific user challenge by it's particular id.

**Pull request merged**